WordPress/src/main/java/org/wordpress/android/ui/posts/PostActionHandler.kt

```diff
@@ -108,45 +110,32 @@ class PostActionHandler(
         }
     }
 
-    private fun restorePost(post: PostModel) {
-        // We need network connection to restore a post
+    private fun moveTrashedPostToDraft(post: PostModel) {
+        /*
+         * We need network connection to move a post to remote draft. We can technically move it to the local drafts
+         * but that'll leave the trashed post in the remote which can be confusing.
+         */
         if (!checkNetworkConnection.invoke()) {
             return
         }
+        post.status = DRAFT.toString()
+        dispatcher.dispatch(PostActionBuilder.newPushPostAction(RemotePostPayload(post, site)))
 
-        criticalPostActionTracker.add(localPostId = LocalId(post.id), criticalPostAction = RESTORING_POST)
-        dispatcher.dispatch(PostActionBuilder.newRestorePostAction(RemotePostPayload(post, site)))
-    }
-
-    fun handlePostRestored(localPostId: LocalId, isError: Boolean) {
-        if (criticalPostActionTracker.get(localPostId) != RESTORING_POST) {
-            /*
-             * This is an unexpected action and either it has already been handled or another critical action has
-             * been performed. In either case, safest action is to just ignore it.
-             */
-            return
-        }
-        val removeFromTracker = {
-            criticalPostActionTracker.remove(localPostId = localPostId, criticalPostAction = RESTORING_POST)
-        }
-        if (isError) {
-            removeFromTracker.invoke()
-            showToast.invoke(ToastMessageHolder(R.string.error_restoring_post, Duration.SHORT))
-            return
+        val localPostId = LocalId(post.id)
+        criticalPostActionTracker.add(localPostId, MOVING_POST_TO_DRAFT)
+        val removeFromCriticalPostActionTracker = {
+            criticalPostActionTracker.remove(localPostId, MOVING_POST_TO_DRAFT)
         }
+
         val snackBarHolder = SnackbarMessageHolder(
-                messageRes = R.string.post_restored,
+                messageRes = R.string.post_moving_to_draft,
                 buttonTitleRes = R.string.undo,
                 buttonAction = {
-                    val post = postStore.getPostByLocalPostId(localPostId.value)
-                    if (post != null) {
-                        removeFromTracker.invoke()
-                        trashPost(post)
-                        showSnackbar.invoke(SnackbarMessageHolder(R.string.post_trashing))
-                    }
+                    removeFromCriticalPostActionTracker.invoke()
+                    trashPost(post)
```

Comment:
As discussed on the call, I've removed the `Undo` action from this Snackbar. This is not only the easiest solution, but maybe the right thing to do regardless of this. This way the user is able to publish the post immediately after it's moved to draft and that might be what they wanted to do in the first place since we don't have a restore post action.